src/Integration/UriExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="UriExtensions.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    internal static class UriExtensions
+    {
+        public static Uri EnsureTrailingSlash(this Uri uri)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException(nameof(uri));
+            }
+
+            var uriString = uri.ToString();
```

Comment:
> Is there a way to avoid .ToString() i.e. using AbsoluteUri
No, AbsoluteUri will include everything, path, query string, fragments, etc...

Speaking of which, what happens (and what _should_ happen) if the user specifies a server URL with these things? e.g., `http://localhost:9000/path/to/somewhere?query=true#fragment`?